BeaKona.AutoInterfaceGenerator/AutoInterfaceSourceGenerator.cs

```diff
@@ -564,7 +564,7 @@ private static List<AutoInterfaceRecord> CollectRecords(GeneratorExecutionContex
 
             foreach (IGrouping<INamedTypeSymbol, IMemberInfo> group in infos.GroupBy(i => i.InterfaceType))
             {
-                List<IMemberInfo> references = group.ToList();
+                List<IMemberInfo> references = group.DistinctBy(x => x.Member.ContainingType).ToList();
```

Comment:
Removing distinct members is not intended by original intention. Holding multiple references is not a bug, but maybe not desired behavior. Collection named `references` is only collection that holds all members by same key-type. References appears to be the same but they are not by two aspects:
1. User can add multiple AutoInterface attributes -> each managing specific aspects of 'forwarding'.. for example:
```cs
[BeaKona.AutoInterfaceTemplate(BeaKona.AutoInterfaceTargets.PropertyGetter, Filter = "Length", Language = "scriban", Body = "return 1;")]
[BeaKona.AutoInterfaceTemplate(BeaKona.AutoInterfaceTargets.Method, Filter = "Print(\\d)?", Body = "LogDebug(nameof({{interface}}.{{name}})); {{expression}};")]
```
first part manages scripts for getters and not methods... second one manages methods and not getters.. I can see this as desirable and undesirable at the same time depending of use case..

2. User can write
```cs
[BeaKona.AutoInterface]
private readonly IPrintable? aspect1 = new PrinterV1();
[BeaKona.AutoInterface(typeof(IPrintable))]
[BeaKona.AutoInterface]
private readonly IPrintable? aspect2 = new PrinterV1();
```
and this will generate calls to both members
```cs
partial class Person
    {
        void IPrintable.Print1()
        {
            this.aspect1!.Print1();
            this.aspect2!.Print1();
        }
```

removing duplicates would lose all entries for every AutoInterface attribute except first one, thus changing main reason I wrote this source generator.
If one or both of such split-depmands are desirable should we add some options (ie. AutoInterfaceAttribute parameters)??

Reply:
The original idea of duplicates removal is to prevent calling the same field method multiple times. I've mistaken with distinction key and just fixed it.

By now I've only tested solution with `AutoInterface` attribute. On my way to test it with `AutoInterfaceTemplate`.

Reply:
Reverted duplication removal idea. I decided, it would be better to provide diagnostic message rather than messing around with specified attributes.
Anyway, it's supposed to be another issue and PR.